lib/mongoid/fields/field_types.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Fields
+
+    # Singleton module which contains a cache for field type definitions.
+    # Custom field types can be configured.
+    module FieldTypes
+      extend self
+
+      # For fields defined with symbols use the correct class.
+      DEFAULT_MAPPING = {
+        array: Array,
+        bigdecimal: BigDecimal,
+        big_decimal: BigDecimal,
+        binary: BSON::Binary,
+        boolean: Mongoid::Boolean,
+        date: Date,
+        datetime: DateTime,
+        date_time: DateTime,
+        decimal128: BSON::Decimal128,
+        double: Float,
+        float: Float,
+        hash: Hash,
+        integer: Integer,
+        object: Object,
+        object_id: BSON::ObjectId,
+        range: Range,
+        regexp: Regexp,
+        set: Set,
+        string: String,
+        stringified_symbol: Mongoid::StringifiedSymbol,
+        symbol: Symbol,
+        time: Time,
+        time_with_zone: ActiveSupport::TimeWithZone
+      }.with_indifferent_access.freeze
+
+      def get(value)
+        value = value.to_sym if value.is_a?(String)
```

Comment:
The mapping is already indifferently accessible, the conversion to symbol is not needed.

Reply:
Good catch, thanks.